arcgis-ios-sdk-samples/Scenes/Realistic lighting and shadows/RealisticLightingAndShadowsViewController.swift

```diff
@@ -0,0 +1,120 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class RealisticLightingAndShadowsViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The label to display the date time set by the slider.
+    @IBOutlet weak var dateTimeLabel: UILabel!
+    /// The slider to change the time of the day.
+    @IBOutlet weak var minuteSlider: UISlider!
+    /// The scene view managed by the view controller.
+    @IBOutlet weak var sceneView: AGSSceneView! {
+        didSet {
+            sceneView.scene = makeScene()
+            sceneView.atmosphereEffect = .realistic
+            sceneView.sunLighting = .lightAndShadows
+        }
+    }
+    
+    // MARK: Instance properties
+    
+    /// A DateComponents struct to encapsulate the minute value from the slider.
+    var dateComponents = DateComponents()
+    /// The Date object of today.
+    let today = Calendar.current.startOfDay(for: Date())
+    /// A date formatter to format date time output.
+    let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.locale = .current
+        formatter.dateStyle = .medium
+        formatter.timeStyle = .short
+        formatter.timeZone = .current
+        return formatter
+    }()
+    
+    // MARK: Instance methods
+    
+    /// Create a scene.
+    ///
+    /// - Returns: A new `AGSScene` object.
+    func makeScene() -> AGSScene {
+        // Create a scene layer from buildings REST service.
+        let buildingsURL = URL(string: "https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_BuildingShells/SceneServer")!
+        let buildingsLayer = AGSArcGISSceneLayer(url: buildingsURL)
+        // Create an elevation source from Terrain3D REST service.
+        let elevationServiceURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+        let elevationSource = AGSArcGISTiledElevationSource(url: elevationServiceURL)
+        let surface = AGSSurface()
+        surface.elevationSources = [elevationSource]
+        let scene = AGSScene(basemap: .topographic())
+        scene.baseSurface = surface
+        scene.operationalLayers.add(buildingsLayer)
+        // Load the scene.
+        scene.load { [weak self] (error) in
+            if let error = error {
+                self?.presentAlert(error: error)
+            } else {
+                let camera = AGSCamera(latitude: 45.54605, longitude: -122.69033, altitude: 941.00021, heading: 162.58544, pitch: 60.0, roll: 0)
+                self?.sceneView.setViewpointCamera(camera)
+            }
+        }
```

Comment:
I'd like to know your opinions on putting load function in this method, or even using the loadable pattern.

I feel like `makeMap()`/`makeScene()` functions should only create an object, and not manipulate UI even in escaping closures. But it also makes sense to load the object on creation, to avoid calling load again somewhere else.

Reply:
It really depends. In this case, there's no reason to explicitly call load. The viewpoint camera can be set on the scene view as soon as it is available. There's no need to wait until the scene is loaded to set the viewpoint camera on the scene view.